cmd/apiKeyValueMapUpload_test.go

```diff
@@ -0,0 +1,96 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRunApiKeyValueMapUpload(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Successfull Api Key Value Map Create Test", func(t *testing.T) {
+		// init
+		config := getDefaultOptions()
+		httpClient := httpMockCpis{CPIFunction: "ApiKeyValueMapUpload", ResponseBody: ``, TestType: "PositiveCase"}
+		// test
+		err := runApiKeyValueMapUpload(&config, nil, &httpClient)
+		// assert
+		if assert.NoError(t, err) {
+			t.Run("check url", func(t *testing.T) {
+				assert.Equal(t, "https://demo/apiportal/api/1.0/Management.svc/KeyMapEntries", httpClient.URL)
+			})
+
+			t.Run("check method", func(t *testing.T) {
+				assert.Equal(t, "POST", httpClient.Method)
+			})
+		}
+	})
+
+	t.Run("Failed case of API Key Value Map Create Test", func(t *testing.T) {
+		// init
+		config := getDefaultOptions()
+
+		httpClient := httpMockCpis{CPIFunction: "ApiKeyValueMapUpload", ResponseBody: ``, TestType: "Negative"}
+
+		// test
+		err := runApiKeyValueMapUpload(&config, nil, &httpClient)
+
+		// assert
+		assert.EqualError(t, err, "HTTP \"POST\" request to \"https://demo/apiportal/api/1.0/Management.svc/KeyMapEntries\" failed with error: 401 Unauthorized")
+	})
+
+	t.Run("Test API Key Value Map payload", func(t *testing.T) {
+		// init
+		config := getDefaultOptions()
+
+		// test
+		payload, err := createJSONPayload(&config)
+		// assert
+		assert.NoError(t, err)
+		assert.NotNil(t, payload)
```

Comment:
Is it not important how the payload looks?

Reply:
since we are simulating HTTP success/failure, its ok to do that i feel.

Reply:
I don't agree. To me it is important that the payload looks as expected, to make sure there is no accidental wrong configuration. Say someone changing your code were to confuse the `name` and the `map_name` fields. It would be great to detect this here, in the unit test, rather than later when the step is used. You might never detect some mismatch like that.

Reply:
i will add assert.equal for payload check